plugins/modules/discovery_intent.py

```diff
@@ -548,6 +593,32 @@ def create_params(self, credential_ids=None, ip_address_list=None):
         if credential_ids is None:
             credential_ids = []
 
+        http_read_credential = self.validated_config[0].get('http_read_credential')
+        http_write_credential = self.validated_config[0].get('http_write_credential')
+        if http_read_credential:
+            if not (http_read_credential.get('password') and isinstance(http_read_credential.get('password'), str)):
+                msg = "Http Read Credential must have a password of type string"
+            if not (http_read_credential.get('username') and isinstance(http_read_credential.get('username'), str)):
+                msg = "Http Read Credential must have a username of type string"
+            if not (http_read_credential.get('password') and isinstance(http_read_credential.get('password'), int)):
+                msg = "Http Read Credential must have port of type integer"
```

Comment:
msg = "The port for the HTTP Read Credential must be of integer type.

Reply:
Addressed